keras_cv/layers/preprocessing/repeated_augmentation.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing.base_image_augmentation_layer import (
+    BaseImageAugmentationLayer,
+)
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class RepeatedAugmentation(BaseImageAugmentationLayer):
+    """RepeatedAugmentation augments each image in a batch multiple times.
+
+        This technique exists to emulate the behavior of stochastic gradient descent within
+        the context of mini-batch gradient descent.  When training large vision models,
+        choosing a large batch size can introduce too much noise into aggregated gradients
+        causing the overall batch's gradients to be less effective than gradients produced
+        using smaller gradients.  RepeatedAugmentation handles this by re-using the same
+        image multiple times within a batch creating correlated samples.
+
+        Args:
+            augmenters: the augmenters to use to augment the image
+            shuffle: whether or not to shuffle the result.  Essential when using an
+                asynchronous distribution strategy such as ParameterServerStrategy.
+
+        Usage:
+        ```python
+        repeated_augment = cv_layers.RepeatedAugmentation(
+            augmenters=[
```

Comment:
I would expect with this kind of functionality that it's somewhat common for the N augmenters to be identical -- is that true?

If so, we should strive to make the API support that without having to pass a list of N of the same object

Reply:
Good point, my thought for this was:

```python
[cv_layers.RandAugment(value_range=(0, 255)] * 8
```

Which works.  but maybe a specialized API surface would be better?  Open to feedback/ideas on this for sure - it is absolutely a common use case.